addons/osfstorage/views.py

```diff
@@ -225,16 +226,16 @@ def osfstorage_create_child(file_node, payload, node_addon, **kwargs):
     if not (name or user) or '/' in name:
         raise HTTPError(httplib.BAD_REQUEST)
 
-    try:
-        # Create a save point so that we can rollback and unlock
-        # the parent record
-        with transaction.atomic():
+    # Create a save point so that we can rollback and unlock
+    # the parent record
+    with transaction.atomic():
+        try:
+            created, file_node = False, parent.find_child_by_name(name, kind=int(not is_folder))
+        except NoResultsFound:
```

Comment:
I think there was some change that made the django exceptions be caught and rethrow the modm version.